internal/repository/repository.go

```diff
@@ -10,6 +10,7 @@ import (
 	"sync"
 
 	"github.com/restic/chunker"
+	"github.com/restic/restic/internal/backend/dryrun"
```

Comment:
**question (non-blocking):** I wonder whether there's an elegant way to avoid the explicit dependency on the dry-run backend here. Using the `backendTestHook` would probably work, but that definitely doesn't count as elegant.

Reply:
IMO the `backendTestHook` would not work as intended. It is applied before the the backend is wrapped by a `CachedBackend`. So, dry-run writes to the repository would not write to the repo, but write to the cache which is not intended.

The problem with the cache is that it depends on the repo ID which is only available after actually reading the config file from the backend. This is why the cache wrapper is added in `repository.go`. Now as the dryrun wrapper should be applied to the already cache-enabled backend, I added this also here.
The alternative would be to allow direct access to the backend of the repository via some `GetBackend` and `SetBackend` methods and then call
```golang 
repo.SetBackend(dryrun.New(repo.GetBackend()))
```
in `cmd_backup.go`. This way we could also separate (most of) the cache logic from the backend.

Reply:
Indeed, you're right, the backend is wrapped with the `CacheBackend` later on.

Making arbitrary modifications to the used backend also doesn't feel particularly elegant, so let's just leave it like it is now.